materialviewpager/src/main/java/com/github/florent37/materialviewpager/MaterialViewPagerAnimator.java

```diff
@@ -303,19 +303,19 @@ public void setColor(int color, int duration) {
         colorAnim.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
             @Override
             public void onAnimationUpdate(ValueAnimator animation) {
-                int colorAlpha = colorWithAlpha((Integer) animation.getAnimatedValue(), lastPercent);
+                final int animatedValue = (Integer) animation.getAnimatedValue();
+                int colorAlpha = colorWithAlpha(animatedValue, lastPercent);
                 mHeader.headerBackground.setBackgroundColor(colorAlpha);
                 mHeader.statusBackground.setBackgroundColor(colorAlpha);
                 mHeader.toolbar.setBackgroundColor(colorAlpha);
                 mHeader.toolbarLayoutBackground.setBackgroundColor(colorAlpha);
                 mHeader.mPagerSlidingTabStrip.setBackgroundColor(colorAlpha);
+
+                //set the new color as MaterialViewPager's color
+                settings.color = animatedValue;
             }
         });
         colorAnim.start();
-
-        //set the new color as MaterialViewPager's color
-        this.settings.color = color;
```

Comment:
Moved this line into Animator Update Listener, since this is starting color for Animator in next tab.